src/mumble/dialogs/PositionalAudioViewer.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "PositionalAudioViewer.h"
+#include "ui_PositionalAudioViewer.h"
+
+#include "PluginManager.h"
+#include "Global.h"
+
+PositionalAudioViewer::PositionalAudioViewer(QWidget *parent) : QDialog(parent), m_ui(new Ui::PositionalAudioViewer) {
+	m_ui->setupUi(this);
+
+	connect(&m_timer, &QTimer::timeout, this, &PositionalAudioViewer::update);
+
+	m_timer.start(100);
+}
+
+PositionalAudioViewer::~PositionalAudioViewer() {
+}
+
+void PositionalAudioViewer::update() {
+	auto pluginManager = Global::get().pluginManager;
+	if (!pluginManager) {
+		return;
+	}
+
+	pluginManager->fetchPositionalData();
+
+	const auto &data = pluginManager->getPositionalData();
+
+	updatePlayer(data);
+	updateCamera(data);
+
+	m_ui->context->setPlainText(data.getContext());
+	m_ui->identity->setPlainText(data.getPlayerIdentity());
+}
+
+void PositionalAudioViewer::updatePlayer(const PositionalData &data) {
+	const auto pos  = data.getPlayerPos();
+	const auto dir  = data.getPlayerDir();
+	const auto axis = data.getPlayerAxis();
+
+	m_ui->playerPosX->setValue(pos.x);
+	m_ui->playerPosY->setValue(pos.y);
+	m_ui->playerPosZ->setValue(pos.z);
+
+	m_ui->playerDirX->setValue(dir.x);
+	m_ui->playerDirY->setValue(dir.y);
+	m_ui->playerDirZ->setValue(dir.z);
+
+	m_ui->playerAxisX->setValue(axis.x);
+	m_ui->playerAxisY->setValue(axis.y);
+	m_ui->playerAxisZ->setValue(axis.z);
+}
+
+void PositionalAudioViewer::updateCamera(const PositionalData &data) {
+	const auto pos  = data.getCameraPos();
+	const auto dir  = data.getCameraDir();
+	const auto axis = data.getCameraAxis();
```

Comment:
`auto` hides variable type

Plus copying this data seems unnecessary. A reference should be just fine :thinking: